core/exchange.go

```diff
@@ -78,31 +79,36 @@ func (ce *Exchange) Get(ctx context.Context, hash libhead.Hash) (*header.Extende
 	log.Debugw("requesting header", "hash", hash.String())
 	block, err := ce.fetcher.GetBlockByHash(ctx, hash)
 	if err != nil {
+		log.Errorw("fetching block by hash", "hash", hash, "err", err)
 		return nil, err
 	}
 
 	comm, vals, err := ce.fetcher.GetBlockInfo(ctx, &block.Height)
 	if err != nil {
+		log.Errorw("fetching block info", "height", &block.Height, "err", err)
```

Comment:
Does the caller log errors? I wonder if we actually could wrap error here and log on caller side.

Reply:
https://github.com/celestiaorg/celestia-node/pull/1961#issuecomment-1481863370